core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -458,6 +461,13 @@ public int register(String subject,
                 + "to generating an ID that is already in use.");
           }
         }
+        for (SchemaValue schemaValue : deletedVersions) {
+          if (schemaValue.getId().equals(schema.getId())) {
+            // Tombstone previous version with the same ID
+            SchemaKey key = new SchemaKey(schemaValue.getSubject(), schemaValue.getVersion());
+            kafkaStore.delete(key);
+          }
+        }
```

Comment:
The tombstoning now happens here in register vs. in `KafkaStoreMessageHandler`.  This removes the need for `guidToDeletedSchemaKeys`